### ISSUE TITLE
Caching

### ISSUE THREAD

**Issue by nerocui** [JitHubApp/JitHubV2#50]
All network calls should be cached for faster UX

**Comment by roke-julian-lockhart** [comment]
> https://github.com/JitHubApp/JitHubV2/issues/50#issue-1708830537

@nerocui, if so, there shall need to be an obvious and ever-present way to clear the cache manually. A refresh button, perhaps?

**Comment by nerocui** [comment]
Yeah that's been on my mind too. My currently WIP displays the cached value and always makes a new call anyway. So the cache is more served as a way to have faster initial UI. So this way I need to let user know the initial UI is "loading" while I fetch for latest data. Perhaps it could be more relaxed on things that don't changes much, like repo name or issue names.

**Comment by roke-julian-lockhart** [comment]
> https://github.com/JitHubApp/JitHubV2/issues/50#issuecomment-2298834378

@nerocui, I like that UX a lot more. It means that whilst it's loading, I can clearly see that, but also not waste time if it's something that I know hasn't changed.
